src/components/views/elements/MemberEventListSummary.js

```diff
@@ -52,88 +52,122 @@ module.exports = React.createClass({
         });
     },
 
-    _getEventSenderName: function(ev) {
-        if (!ev) {
-            return 'undefined';
+    _renderNameList: function(users) {
+        if (users.length === 0) {
+            return null;
         }
-        return ev.sender.name || ev.event.content.displayname || ev.getSender();
-    },
+        let originalNumber = users.length;
 
-    _renderNameList: function(events) {
-        if (events.length === 0) {
-            return null;
+        users = users.slice(0, this.props.summaryLength);
+
+        let remaining = originalNumber - this.props.summaryLength;
+        if (remaining < 0) {
+            remaining = 0;
         }
-        let originalNumber = events.length;
-        events = events.slice(0, this.props.summaryLength);
-        let lastEvent = events.pop();
-
-        let names = events.map((ev) => {
-            return this._getEventSenderName(ev);
-        }).join(', ');
-
-        let lastName = this._getEventSenderName(lastEvent);
-        if (names.length === 0) {
-            // special-case for a single event
-            return lastName;
+        let other = " other" + (remaining > 1 ? "s" : "");
+
+        return this._renderCommaSeparatedList(users, remaining) + (remaining ? ' and ' + remaining + other : '');
+    },
+
+    // Test whether the first n items repeat for the duration
+    // e.g. [1,2,3,4,1,2,3] would resolve true for n = 4
+    _isRepeatedSequence: function(transitions, n) {
+        let count = 0;
+        for (let i = 0; i < transitions.length; i++) {
+            if (transitions[i % n] !== transitions[i]) {
+                return null;
+            }
         }
+        return true;
+    },
 
-        let remaining = originalNumber - this.props.summaryLength;
-        if (remaining > 0) {
-            //  name1, name2, name3, and 100 others
-            return names + ', ' + lastName + ', and ' + remaining + ' others';
+    _renderCommaSeparatedList(items, disableAnd) {
```

Comment:
rather than `disableAnd`, it might be preferable to have a `itemLimit` param, and move the 'and x others' logic in here - this making it reusable.

```
const remaining = itemLimit === undefined ? 0 : Math.max(items.length - itemLimit, 0);
if (items.length === 0) {
    return "";
} else if (items.length === 1) {
    return items[0];
} else if (remaining) {
    items = items.slice(0, itemLimit);
    const other = " other" + (remaining > 1 ? "s" : "");
    return items.join(', ') + ' and ' + remaining + other;
} else {
    let last = items.pop();
    return items.join(', ') + ' and ' + last;
}
```